embedded/store/ongoing_tx.go

```diff
@@ -182,6 +185,24 @@ func (tx *OngoingTx) Set(key []byte, md *KVMetadata, value []byte) error {
 	return nil
 }
 
+func (tx *OngoingTx) AddKVConstraint(c WriteConstraint) error {
```

Comment:
probably it's better to have a single naming for this constraints both in method and type names

Reply:
Changed to Precondition everywhere